packages/datadog-plugin-sharedb/src/index.js

```diff
@@ -0,0 +1,125 @@
+'use strict'
+
+/**
+ * @description The enum values in this map are not exposed from ShareDB, so the keys are hard-coded here.
+ * The values were derived from: https://github.com/share/sharedb/blob/master/lib/client/connection.js#L196
+ */
+const READABLE_ACTION_NAMES = {
+  hs: 'handshake',
+  qf: 'query-fetch',
+  qs: 'query-subscribe',
+  qu: 'query-unsubscribe',
+  bf: 'bulk-fetch',
+  bs: 'bulk-subscribe',
+  bu: 'bulk-unsubscribe',
+  f: 'fetch',
+  s: 'subscribe',
+  u: 'unsubscribe',
+  op: 'op',
+  nf: 'snapshot-fetch',
+  nt: 'snapshot-fetch-by-ts',
+  p: 'presence-broadcast',
+  pr: 'presence-request',
+  ps: 'presence-subscribe',
+  pu: 'presence-unsubscribe'
+}
+
+function getReadableActionName (action) {
+  const actionName = READABLE_ACTION_NAMES[action]
+  if (actionName === undefined) {
+    return action
+  }
+  return actionName
+}
+
+function getReadableResourceName (readableActionName, collection, query) {
+  if (collection) {
+    readableActionName += ' ' + collection
+  }
+  if (query) {
+    readableActionName += ' ' + JSON.stringify(sanitize(query))
+  }
+  return readableActionName
+}
+
+function sanitize (input) {
+  const output = {}
+
+  if (!isObject(input) || Buffer.isBuffer(input)) return '?'
+
+  for (const key in input) {
+    if (typeof input[key] === 'function') continue
+
+    output[key] = sanitize(input[key])
+  }
+
+  return output
+}
+
+function isObject (val) {
+  return typeof val === 'object' && val !== null && !(val instanceof Array)
+}
+
+function wrapCallback (config, tracer, request, span, done) {
+  return tracer.scope().bind((err, res) => {
+    if (err) {
```

Comment:
It's also possible to do `span.setTag('error', err)` which also has a built-in check if `err` has a value or not.

Reply:
There is already an `if` in `setTag` so this is redundant.